stdlib/LinearAlgebra/src/givens.jl

```diff
@@ -250,20 +250,23 @@ function givensAlgorithm(f::Complex{T}, g::Complex{T}) where T<:AbstractFloat
     return cs, sn, r
 end
 
-# enable for unitful quantities
-function givensAlgorithm(f::T, g::T) where T
-    fs = f / oneunit(T)
-    gs = g / oneunit(T)
-    typeof(fs) === T && typeof(gs) === T &&
-    !isa(fs, Union{AbstractFloat,Complex{<:AbstractFloat}}) &&
-    throw(MethodError(givensAlgorithm, (fs, gs)))
-
-    c, s, r = givensAlgorithm(fs, gs)
-    return c, s, r * oneunit(T)
+# From Janovská, D., & Opfer, G. (2003). Givens’ Transformation Applied to Quaternion
+# Valued Vectors. BIT Numerical Mathematics, 43(5), 991–1002.
+# doi:10.1023/b:bitn.0000014561.58141.2c
+function givensAlgorithm(f::Number, g::Number)
+    nrm = hypot(f, g)
+    c = abs(f) / nrm
+    s, u = if iszero(f)
+        -one(first(promote(f, g))), -g
```

Comment:
I just noticed that this is inconsistent with the existing definition for `Float64`:
```julia
julia> using LinearAlgebra

julia> LinearAlgebra.givensAlgorithm(0.0, 1.0)
(0.0, 1.0, 1.0)
```
It seems to be consistent with the existing definitions we would need something like
```suggestion
        one(first(promote(f, g))), g
```

Reply:
I'm not sure it's worth trying to match the values of `c` and `s` from the two versions translated from LAPACK. Then we'd essentially have to reimplement the version for complex numbers and I'm not sure if it necessarily works for quaternions. In any case, which of the allowed values of `c` and `s` doesn't really matter. It's always the result of applying the rotation that matters and it isn't affected by this choice (except for rounding).

Reply:
Thought a bit more about this and concluded that you are right here. The fallback would have to make the same sign decisions as the specialized version so the approach of this PR can't work. We should probably try to make a version of the complex version that isn't restricted to `Complex32/64`.